tests/test_communication_manager.py

```diff
@@ -0,0 +1,163 @@
+import tests.context
+
+import pytest
+import time
+
+from core.communication_manager import CommunicationManager
+from core.utils.event_types     import RawEventTypes, ActionableEventTypes
+from core.runner                import DMLRunner
+from core.scheduler             import DMLScheduler
+from custom.keras               import get_optimizer
+from models.keras_perceptron    import KerasPerceptron
+from core.utils.dmljob          import DMLJob, serialize_job, deserialize_job
+from core.configuration         import ConfigurationManager
+
+def make_model_json():
+    m = KerasPerceptron(is_training=True)
+    model_architecture = m.model.to_json()
+    model_optimizer = get_optimizer(m.model)
+    model_json = {
+        "architecture": model_architecture,
+        "optimizer": model_optimizer
+    }
+    return model_json
+
+def make_initialize_job(model_json):
+    initialize_job = DMLJob(
+        "initialize",
+        model_json,
+        "keras",
+        hyperparams=make_hyperparams(),
+        label_column_name='label'
+    )
+    return initialize_job
+
+def make_hyperparams():
+    hyperparams = {
+        'averaging_type': 'data_size',
+        'batch_size': 4,
+        'epochs': 1,
+        'split': 0.004,
+    }
+    return hyperparams
+
+config_manager = ConfigurationManager()
+config_manager.bootstrap(
+    config_filepath='tests/artifacts/communication_manager/configuration.ini'
+)
+
+def test_communication_manager_can_initialize_and_train_model():
```

Comment:
If you can't split this up, can you identify which code in the test corresponds to which task it's trying to accomplish? Like a comment above a code block could be `# (1), (2), (3)

Reply:
gotcha

Reply:
jk i can't it all happens in one line

Reply:
> one line

Then can you label exactly where in the code it happens? This can simply be adding numbers to the documentation in the code so that the exact order of operations is more visible.